src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetSourcesService.cs

```diff
@@ -39,11 +39,21 @@ public async ValueTask<IReadOnlyList<PackageSource>> GetPackageSourcesAsync(Canc
             return packageSources.PackageSourceProvider.LoadPackageSources().ToList();
         }
 
-        public async ValueTask SavePackageSourcesAsync(IReadOnlyList<PackageSource> sources, CancellationToken cancellationToken)
+        public async ValueTask SavePackageSourcesAsync(IReadOnlyList<PackageSourceTransaction> sources, CancellationToken cancellationToken)
         {
             var packageSources = await ServiceLocator.GetInstanceAsync<ISourceRepositoryProvider>();
             Assumes.NotNull(packageSources);
-            packageSources.PackageSourceProvider.SavePackageSources(sources);
+
+            var packageSources2 = packageSources.PackageSourceProvider as IPackageSourceProvider2;
+            if(packageSources2 != null)
+            {
+                packageSources2.SavePackageSources(sources);
+            }
+            else
+            {
+                packageSources.PackageSourceProvider.SavePackageSources(sources.Select(s => s.PackageSource).ToList());
```

Comment:
>packageSources [](start = 16, length = 14)

in which cases will we hit this code? clients before this change?
if so, do we need this code? for how long? etc...

Reply:
IPackageSourceProvider is an extension point so it should be supported, as far as deprecating or plans to unify I am not sure if that would be a current goal.